File: java/src/main/java/testingil/webinar/tddapi/sfinal/Navigator.java
package testingil.webinar.tddapi.sfinal;

public class Navigator {

	private DistanceProvider distProvider;
	private Location startingPoint;
	private Location endPoint;

	public Navigator(DistanceProvider distProvider) {
		this.distProvider = distProvider;
	}

	public Distance getDistanceFromDestination()	{
		return distProvider.getDistance(startingPoint, endPoint);
	}

	public void setDestination(Location dest) {
		this.endPoint = dest;
	}

	public void setStartingPoint(Location startingPoint) {
		this.startingPoint = startingPoint;
	}

}

File: Python/s10_refactor_provider/navigator.py
from s10_refactor_provider.distance import Distance


class Navigator:
    def __init__(self, starting_point, dist_provider):
        self.dist_provider = dist_provider
        self.starting_point = starting_point

    def getDistanceFromDestination(self):
        return self.dist_provider.getDistance(self.starting_point, self.starting_point)

    def setDestination(self, dest):
        pass
